modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskService.java

```diff
@@ -17,11 +19,13 @@
 public interface TaskService {
 
     /**
-     * Saves the given task in the database.
+     * Saves the given task in the database and returns the list
+     * of errors that will occur when enabling task.
      *
      * @param task  the task to be saved, not null
+     * @return list of task errors
      */
-    void save(final Task task);
+    Set<TaskError> save(final Task task);
```

Comment:
We should also improve the error alert that is displayed when clicking enable in the task list view. Currently it doesn't list the errors, just a cryptic 'can't do this' message.